.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,53 @@
+name: Build and Test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types:
+      - opened
+      - synchronize
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
```

Comment:
<img width="1068" alt="Screen Shot 2023-10-30 at 10 21 55 PM" src="https://github.com/twang7310/rate-my-cse/assets/145406076/7b50afa5-d734-4183-9327-c0d6ce0cbbcc">

These two warnings show up when running this workflow. I believe it's because, while you specify node-version 16, the actions version v2 may be outdated. 

This is just from Googling and ChatGPTing, but I think changing all these actions/__@v2 to v3 or v4 (most recent) would fix this. 

Though this is probably trivial in the long run.

Reply:
Fixed